File: src/MrHotel/WebApp/src/app/services/report-manager/data/product-offset-data.ts
import { Guid } from "@customTypes/.";

export class ProductOffsetData {
  public constructor(
    private readonly _productId: Guid,
    private readonly _quantity: number,
  ) {}

  public get id(): Guid {
    return this._productId;
  }

  public get quantity(): number {
    return this._quantity;
  }

  public toJSON(): Record<string, string> {
    return {
      productId: this._productId,
      quantity: this._quantity.toString(),
    };
  }
}

File: src/MrHotel/ApiService/Reports/Data/ProductOffsetData.cs
namespace MrHotel.ApiService.Reports.Data;

using MrHotel.Database.Entities.Reports;

public record ProductOffsetData(
    Guid ProductId,
    int Quantity)
{
    public ProductOffset ToProductOffset()
    {
        return new() { Quantity = this.Quantity, ProductId = this.ProductId };
    }
}
